frontend/server/src/main/java/com/amazonaws/ml/mms/util/ConfigManager.java

```diff
@@ -368,6 +374,24 @@ private static String getCanonicalPath(String path) {
         return getCanonicalPath(new File(path));
     }
 
+    private static int getAvailableGpu() {
+        try {
+            Process process =
+                    Runtime.getRuntime().exec("nvidia-smi --query-gpu=index --format=csv");
```

Comment:
This will only work for Linux machines. 
Might not work for other OS Types.

Reply:
Why you think this won't work on other OS?

Reply:
An example : 
https://devtalk.nvidia.com/default/topic/839954/cuda-programming-and-performance/command-line-gpu-load-display-on-os-x/post/4563535/#4563535

Reply:
From a more official source : https://developer.nvidia.com/nvidia-system-management-interface

```
NVIDIA-smi ships with NVIDIA GPU display drivers on Linux, and with 64bit Windows Server 2008 R2 and Windows 7. Nvidia-smi can report query information as XML or human readable plain text to either standard output or a file. For more details, please refer to the nvidia-smi documentation.
```

Reply:
@piyushghai : Do you know of any tool which can be used across platforms?

Reply:
None that I'm aware of.